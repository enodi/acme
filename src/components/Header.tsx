import React from "react";

import { ReactComponent as Hamburger } from "../assets/hamburger.svg";
import { userDataFormat } from "../models";

interface HeaderProps {
  toggleSidebar: () => void;
  selectedItem: userDataFormat;
}

const Header: React.FC<HeaderProps> = ({ toggleSidebar, selectedItem }) => (
  <header className="header">
    <Hamburger className="header__icon" onClick={toggleSidebar} />
    <span>
      {selectedItem && selectedItem.firstName}{" "}
      {selectedItem && selectedItem.lastName}
    </span>
  </header>
);

export default Header;
