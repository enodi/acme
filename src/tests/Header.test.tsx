import React from "react";
import { shallow } from "enzyme";

import Header from "../components/Header";
import { selectedItem } from "./fixtures";

let toggleSidebar, wrapper: any;
beforeEach(() => {
  toggleSidebar = jest.fn();
  wrapper = shallow(
    <Header toggleSidebar={toggleSidebar} selectedItem={selectedItem} />
  );
});

test("should render Header component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should find text rendered on header", () => {
  expect(wrapper.find("span").text()).toEqual("Kirby Rotte");
});
