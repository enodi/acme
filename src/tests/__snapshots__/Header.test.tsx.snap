// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header
  className="header"
>
  <ForwardRef(SvgHamburger)
    className="header__icon"
    onClick={[MockFunction]}
  />
  <span>
    Kirby
     
    Rotte
  </span>
</header>
`;
